sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -163,6 +249,16 @@ case class TransformWithStateExec(
   override protected def doExecute(): RDD[InternalRow] = {
     metrics // force lazy init at driver
 
+    timeoutMode match {
+      case ProcessingTime =>
+        require(batchTimestampMs.nonEmpty)
```

Comment:
nit: internal error? I'm OK to use require and expect that surrounding code will convert the exception, but probably better to provide the message as well.

Reply:
Done